src/api/transactions.ts

```diff
@@ -16,16 +18,21 @@ export const fetchTransactions = async (
 ): Promise<PagedResult<Transaction>> => {
   const filterKeys = ['block', 'hash', 'address'];
   const { startAt = 0, endAt = 9, term, keys = [] } = params || {};
-  const requestKeys = term ? (keys.length ? keys : filterKeys) : ['$key'];
+  let requestKeys = ['$key'];
+  if (term) {
+    if (term.indexOf('DAG') === 0) {
```

Comment:
If hash begins with DAG then we can assume it is an address and avoid requesting the others.

Reply:
You could also use `String.prototype.startsWith`.

https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith